### DISCUSSION TITLE
How do you schedule BIRT Reports and send by email

### DISCUSSION THREAD

**Discussion by FinnLyn** [eclipse-birt/birt#1968] [Q&A] [answered]
Hi,

I have searched a while for an automated way to invoke a BIRT Report and send the result in an Email as an attached file (pfd / csv).
But I haven't found the solution yet...
I would have been nice, if this function was build in the BIRT Reporting tool!

Do anyone know, it this could be done?

**Comment by hvbtup** [comment, marked as answer]
The BIRT open source project does not contain a fully-featured report server.
But it shouldn't be to difficult to develop your own solution, suited to your needs.
For example, if you want to send a report as an Email attachment, you probably want to control the message subject, body and attachment name, too.

Once the BIRT engine is initialized, executing a report is quite easy.
Take a look at the source file engine\org.eclipse.birt.report.engine\src\org\eclipse\birt\report\engine\api\ReportRunner.java and look for the method runAndRenderReport() there. You can use this code as a template.

For sending emails with SMTP, there are certainly several Java libraries available. However, I don't know, because I use Python for everything except the report generation itself.

**Reply by FinnLyn** [reply]
Thank you for your reply, Henning.
I will try it out, and see if I can make it work, and get back with the result :)

Best regards, Finn.
